clippy_lints/src/swap.rs

```diff
@@ -172,12 +172,13 @@ fn check_manual_swap(cx: &LateContext<'_>, block: &Block<'_>) {
     }
 }
 
+#[allow(clippy::too_many_lines)]
 /// Implementation of the `ALMOST_SWAPPED` lint.
 fn check_suspicious_swap(cx: &LateContext<'_>, block: &Block<'_>) {
-    for w in block.stmts.windows(2) {
+    for [first_, second_] in block.stmts.array_windows() {
         if_chain! {
-            if let StmtKind::Semi(first) = w[0].kind;
-            if let StmtKind::Semi(second) = w[1].kind;
+            if let StmtKind::Semi(first) = first_.kind;
+            if let StmtKind::Semi(second) = second_.kind;
             if first.span.ctxt() == second.span.ctxt();
             if let ExprKind::Assign(lhs0, rhs0, _) = first.kind;
             if let ExprKind::Assign(lhs1, rhs1, _) = second.kind;
```

Comment:
Have another look at the comment https://github.com/rust-lang/rust-clippy/pull/8945#issuecomment-1192657937

We shouldn't be checking the `StmtKind` here, rather in a `parse` function that detects both cases. `ExprOrLocal` would be something along the lines of

```rust
enum ExprOrLocal<'a> {
    Expr(&'a Expr<'a>),
    Local(Ident),
}
```